java/runtime/src/main/java/org/ray/runtime/WorkerContext.java

```diff
@@ -79,48 +50,43 @@ public TaskId getCurrentTaskId() {
   public void setCurrentTask(TaskSpec task, ClassLoader classLoader) {
     if (runMode == RunMode.CLUSTER) {
       Preconditions.checkState(
-              Thread.currentThread().getId() == mainThreadId,
-              "This method should only be called from the main thread."
+          Thread.currentThread().getId() == mainThreadId,
+          "This method should only be called from the main thread."
       );
     }
 
     Preconditions.checkNotNull(task);
-    this.currentTaskId.set(task.taskId);
-    this.currentJobId = task.jobId;
-    taskIndex.set(0);
-    putIndex.set(0);
-    this.currentTask.set(task);
+    ByteBuffer info = RayletClientImpl.convertTaskSpecToFlatbuffer(task);
+    nativeSetCurrentTask(nativeWorkerContext, info, info.position(), info.remaining());
     currentClassLoader = classLoader;
   }
 
   /**
    * Increment the put index and return the new value.
    */
   public int nextPutIndex() {
```

Comment:
I think most of these are not going to be used, or at least not in common path? as submitting & executing tasks are done in `CoreWorker` interface which internally manages the index stuff.